### DISCUSSION TITLE
Markdown support in the `@deprecated` message.

### DISCUSSION THREAD

**Discussion by naomieow** [gleam-lang/gleam#3266] [Ideas & suggestions]
Markdown support in the `@deprecated` message for when it generates in the documentation. This would allow something like this:
```gleam
/// Set the auth token for authentication - this is in a query
@deprecated("This authentication option is deprecated for removal in Gitea 1.23. use [`with_auth_header_token`](#with_auth_header_token) instead.")
pub fn with_auth_token(settings: Settings, token token: String) -> Settings {
  Settings(..settings, token: Some(token))
}
```

**Comment by lpil** [comment]
How would this work given that this message is also included in the command line, in the language server, etc?

**Reply by Octaeon** [reply]
Although this discussion is really old, I support this idea.

The simplest solution would be to use the string as markdown when generating the documentation, and sanitize it when writing to the command line or sending to the code editor through the LSP.

More generally, when I say *sanitize*, I mean that if the `String` is going to be displayed in a context that does not support markdown, then traverse the String, and upon encountering a markdown signifier (such as a link as shown in the original post), parse it to extract what should be shown, and replace that segment with it.

> I imagine it should be possible to achieve using regex, at least for links like the one above - look for any substring matching the pattern `\[.*\]\(.*\)`, and replace it with whatever is inside the square brackets (this won't work, I'm not going to go and write a regex pattern for this right now).

Lastly, the user should be able to control where the resulting `sanitize` function is called when using the `gleam` command.

Specifically, if using the command in a terminal that supports markdown (idk if something like that exists, but I'm guessing it does), disable the `sanitize` function being called when writing to terminal.
Or if using a code editor that does not support markdown (nano, vim, helix, idk - there must exist some editor that can integrate with an LSP but doesn't render markdown), enable using the `sanitize` function on the LSP messages.
